ops/jenkins/bfd-run-synthea-generation/Jenkinsfile

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env groovy
+
+pipeline {
+  agent {
+    kubernetes {
+      defaultContainer 'docker'
+      yaml """
+apiVersion: v1
+kind: Pod
+spec:
+  serviceAccount: bfd
+  volumes:
+  - name: docker-socket
+    emptyDir: {}
+  containers:
+  - name: docker
+    image: docker:20.10-git
+    command:
+    - sleep
+    args:
+    - 99d
+    volumeMounts:
+    - name: docker-socket
+      mountPath: /var/run
+    tty: true
+  - name: docker-daemon
+    image: docker:20.10-dind
+    securityContext:
+      privileged: true
+    volumeMounts:
+    - name: docker-socket
+      mountPath: /var/run
+    resources:
+      limits:
+        memory: 16384Mi
+        cpu: 8000m
+      requests:
+        memory: 16384Mi
+        cpu: 8000m
+"""
+    }
+  }
+
+  parameters {
+    string(name: 'NUM_BENES',
+           defaultValue: '10000',
+           description: 'The number of synthetic beneficiaries to generate')
+
+    string(name: 'NUM_FUTURE_MONTHS',
+           defaultValue: '0',
+           description: 'The number of months into the future that claim lines should be '
+                        + 'generated. If specified, future claims will be split in the '
+                        + 'generated output to allow for proper loading')
+
+    booleanParam(name: 'UPDATE_END_STATE_S3',
+           defaultValue: false,
+           description: 'Specifies whether the main end_state.properties stored in the synthea S3 '
+                        + 'bucket should be updated with the end_state.properties '
+                        + 'of this run. If false, the main end_state.properties is left as-is. '
+                        + 'If true, the main end_state.properties is replaced. Regardless of this '
+                        + 'parameter\'s value, the starting and final end_state.properties will '
+                        + 'be available in the generated output directory uploaded to S3')
+  }
```

Comment:
This should probably default to true; it's more likely we're generating files to be loaded than not (or regenerating a bad batch, the other possible situation I could see wanting this to be false)

Reply:
That's fine with me. I only defaulted it to `false` as a safeguard against accidental "full" generations. Even if we _do_ accidentally keep this `true` when we don't want it to be (i.e. we're testing the pipeline) the original `end_state.properties` is always uploaded as part of the generated output to the `bfd-mgmt-synthea` S3 bucket so we can just replace the `end_state.properties` manually with it.

Reply:
I've updated this parameter to default to `true`, thus defaulting to always updating the previous `end_state.properties` with the new end state.